### ISSUE TITLE
Check cache against users locale to ensure we are serving the proper strings

### ISSUE THREAD

**Issue by micahmills** [DiscipleTools/disciple-tools-theme#903]
If you are running a DT instance with users in multiple languages data gets cached in language 1 and served to those in language 2 because there is no check for locale before pulling from the cache. 

An example of this is the get_custom_fields_settings function on line 682 of contacts-post-type.php. we are loading from cache when it is available, but we don't check if that cache is in the appropriate language for the user. 

I am not super familiar with the wp_cache functions and if there is a native way to cache data based on the locale, but checking the users locale vs the cached data's locale seems like the logical solution.

**Comment by corsacca** [comment]
The cache is per request, so the next request will set up it's own cache.
The issue likely stems from getting the fields before the user locale is set. I'm fixing this in the contacts.php and groups.php in the strings branch.
If we have a function that switches user (maybe to send an email) we would want to user the $load_from_cache = false parameter to get the updated fields.
Did you find a scenario where the wrong language is being used?

**Comment by micahmills** [comment]
I noticed the problem in 2 places. In my local environment the cache was mismatched but this is likely because I was switching languages back and forth, this would not be a normal use case for most users.

The other place I noticed it was on our live server, because the host (Wp-engine) has caching on by default and it seems that it is pulling the cache with the wrong language. This use case is probably more common since many hosts use some form of caching or another.
